python/ray/serve/scripts.py

```diff
@@ -529,14 +534,16 @@ def run(
             "need to call `ray.init` in your code when using `serve run`."
         )
 
-    http_options = {"location": "EveryNode"}
+    proxy_location = None
+    http_options = None
     grpc_options = gRPCOptions()
     # Merge http_options and grpc_options with the ones on ServeDeploySchema.
     if is_config and isinstance(config, ServeDeploySchema):
-        config_http_options = config.http_options.dict()
-        http_options = {**config_http_options, **http_options}
+        proxy_location = config.proxy_location
+        http_options = config.http_options.dict()
```

Comment:
if the core issue we are trying to solve is that serve run does not pick proxy_location from config, then is the following sufficient to fix that?

```python
    http_options = {"location": "EveryNode"}
    grpc_options = gRPCOptions()
    # Merge http_options and grpc_options with the ones on ServeDeploySchema.
    if is_config and isinstance(config, ServeDeploySchema):
        http_options["location"] = config.proxy_location
        config_http_options = config.http_options.dict()
        http_options = {**config_http_options, **http_options}
        grpc_options = gRPCOptions(**config.grpc_options.dict())

    client = _private_api.serve_start(
        http_options=http_options,
        .....
```

Reply:
makes sense, then let's make this change